### DISCUSSION TITLE
AWS Lambda Snapstart has the limited role?

### DISCUSSION THREAD

**Discussion by lgtminshi** [firecracker-microvm/firecracker#3329] [General]
Based on CRIU, snapshot technolygy has developed long long ago. AWS Lambda recently released snapstart. 
It alleviates cold start problems but still doesn't solve the problem, specially for performance-sensitive scenarios. Maybe bcoz of this reason, this feature is good but not that nice though AWS Lambda here recently choose to publish.

**Comment by xmarcalx** [comment]
Hi @lgtminshi, 
thanks for opening this discussion.

I am not sure I am fully understanding the question. Can you expand on it?

Snapshot as technology is present from years in the ecosystem, you can imagine that it is even present in XEN. 
https://xapi-project.github.io/xen-api/snapshots.html

Firecracker on boarded on a journey to support it as a feature to manage our microVMs as well and AWS Lambda decided to use this feature to provide SnapStart to their customers and starting up faster lambda functions.

More details available here:
-	https://aws.amazon.com/blogs/compute/starting-up-faster-with-aws-lambda-snapstart/
-	https://aws.amazon.com/blogs/aws/new-accelerate-your-lambda-functions-with-lambda-snapstart/

**Reply by lgtminshi** [reply]
hi, @xmarcalx;
thanks for your reply. However, the snapshot feature has been available for a long time. Why does Lambda integrate this capability only now?In my heart, AWS is always at the forefront of trends. This late integration（Lambda） makes me question the value of this feature.
It just alleviates the code start problem in serverless, but still not cure the cold start problem. Actually, It has been published on the heavyweight keynote. As a CS student, I can't understand this logic(it's not that important but published on the keynote... quite illogical :-#）

PS:
English is not my mother tongue, I try to explain my question. If not sure fully understanding the question, pls let know. I will give my next try~
